freespace2/freespace.cpp

```diff
@@ -4281,6 +4281,13 @@ void game_set_frametime(int state)
 	if (Frametime > MAX_FRAMETIME)	{
 #ifndef NDEBUG
 		mprintf(("Frame %2i too long!!: frametime = %.3f (%.3f)\n", Framecount, f2fl(Frametime), f2fl(debug_frametime)));
+
+		// If the frame took more than 5 seconds, assume we're tracing through a debugger.  If timestamps are running, correct the elapsed time.
+		if (!timestamp_is_paused() && f2fl(Frametime) > 5.0f) {
+			auto delta_timestamp = timestamp() - Last_frame_timestamp;
```

Comment:
One thing I noticed is that `Last_frame_timestamp` isn't initialized when it's declared, so technically it's not set until after it's used here. If there is a long enough startup time such that this is hit on the first call then it could have unpredictable results.

Most people probably won't have to worry about that but profiling tools can slow things down enough to cause a problem.

Reply:
Good catch.  I initialized it.